### ISSUE TITLE
[nri-bundle] Cannot set newrelic-logging.lowDataMode when any globals are set

### ISSUE THREAD

**Issue by booya** [newrelic/helm-charts#553]
**Bug description**

We have been trying to reduce the volume of data we are sending to new relic recently. As such, we have been looking at low data mode, which looks like it should help a lot with our logging volume since we likely don't need all the metadata included with the regular configuration.  While trying to enable this, I noticed it was overwriting our existing transformations in the nri-promethus values.  I see in the documentation that this is not supported, so I figured I would try to add the prefixes from the static/lowdatamode.yaml configs. into our custom transformations.  This appears to be working.

To work around the issue, I've left global.lowDataMode and nri-prometheus.lowDataMode unset, and attempting to set this via the newrelic-logging.lowDataMode value.  However after installing the chart with these values, I noticed the logging configmap is not including the additional entries for low data mode.  I have traced the problem to the _helpers.tpl file:

```yaml
{{- define "newrelic.lowDataMode" -}}
{{- if .Values.global }}
  {{- if .Values.global.lowDataMode }}
    {{- .Values.global.lowDataMode -}}
  {{- end -}}
{{- else if .Values.lowDataMode }}
  {{- .Values.lowDataMode -}}
{{- end -}}
{{- end -}}
```

Since we do have global values enabled to set the cluster name and the license key, it's not reaching the "else if" block here.  I cannot enable global.lowDataMode as that would overwrite the transformations, which ends up seriously increasing our metric volume since we are already using the ignore_metrics option to remove a fairly noisy set of metrics from istio that we are not using.

**Version of Helm and Kubernetes**

Helm  v3.1.2
Kubernetes v1.20.9-gke.1001

**Which chart?**

We are using the nri-bundle chart to install nri-prometheus and nri-logging which support low data mode. I see the pixie chart has the same _helpers.tpl file, so I assume that chart has the same problem, but we are not installing that one.

**What happened?**

Setting `newrelic-logging.lowDataMode=true` does not apply the low data mode settings to the fluent-bit configmap

**What you expected to happen?**

Setting `newrelic-logging.lowDataMode=true` should apply the low data mode settings to the fluent-bit configmap even when `global.lowDataMode=false`, or if any other global flags are specified (eg global.cluster, global.licenseKey)

**How to reproduce it?**

```bash
helm upgrade --install newrelic-bundle newrelic/nri-bundle \
  --namespace=newrelic \
  --set global.licenseKey=${LICENSE_KEY} \
  --set global.cluster=${CLUSTER_NAME} \
  --set newrelic-logging.lowDataMode=True
```

**Anything else we need to know?**

Perhaps, but they are best left for a different forum :)

**Comment by booya** [comment]
I haven't had time to test this yet, but looking at some of the other definitions in the _helpers.tpl file, this issue may affect other values where there is a global and chart override.  Perhaps something like this would work?

```
{{- define "newrelic.lowDataMode" -}}
{{- if .Values.global }}
  {{- if .Values.global.lowDataMode }}
    {{- .Values.global.lowDataMode -}}
  {{- else if .Values.lowDataMode }}
    {{- .Values.lowDataMode -}}
  {{- end -}}
{{- else if .Values.lowDataMode }}
  {{- .Values.lowDataMode -}}
{{- end -}}
{{- end -}}
```

**Comment by kang-makes** [comment]
Thanks for your patience. It was fixed in #632 a couple of days ago :)

As your problem was with the `lowDataMode` flag I am closing this issue. We are aware of other globals that supersede local variables but we are taking care of that in other PR with different issues to each problem we find.
